src/ol_infrastructure/applications/concourse/__main__.py

```diff
@@ -0,0 +1,532 @@
+            "Action": [
+                "s3:GetObject",
+                "s3:PutObject",
+                "s3:DeleteObject",
+                "s3:ListBucket*",
+            ],
+            "Resource": [
+                "arn:aws:s3:::ocw-content*",
+                "arn:aws:s3:::ocw-content*/*",
+                "arn:aws:s3:::ol-eng-artifacts" "arn:aws:s3:::ol-eng-artifacts/*",
+            ],
+        },
+    ],
+}
+
+concourse_iam_policy = iam.Policy(
+    "cicd-iam-permissions-policy",
+    path=f"/ol-infrastructure/iam/cicd-{stack_info.env_suffix}/",
+    policy=lint_iam_policy(concourse_iam_permissions, stringify=True),
+    name_prefix=f"cicd-policy-{stack_info.env_suffix}",
+    tags=aws_config.tags,
+)
+
+# IAM and instance profile
+concourse_instance_role = iam.Role(
+    f"concourse-instance-role-{stack_info.env_suffix}",
+    assume_role_policy=json.dumps(
+        {
+            "Version": "2012-10-17",
+            "Statement": {
+                "Effect": "Allow",
+                "Action": "sts:AssumeRole",
+                "Principal": {"Service": "ec2.amazonaws.com"},
+            },
+        }
+    ),
+    path="/ol-applications/concourse/role/",
+    tags=aws_config.tags,
+)
+
+iam.RolePolicyAttachment(
+    f"concourse-describe-instance-role-policy-{stack_info.env_suffix}",
+    policy_arn=policy_stack.require_output("iam_policies")["describe_instances"],
+    role=concourse_instance_role.name,
+)
+
+iam.RolePolicyAttachment(
+    f"concourse-route53-role-policy-{stack_info.env_suffix}",
+    policy_arn=policy_stack.require_output("iam_policies")["route53_odl_zone_records"],
+    role=concourse_instance_role.name,
+)
+
+iam.RolePolicyAttachment(
+    "concourse-cicd-permissions-policy",
+    policy_arn=concourse_iam_policy.arn,
+    role=concourse_instance_role.name,
+)
+
+concourse_instance_profile = iam.InstanceProfile(
+    f"concourse-instance-profile-{stack_info.env_suffix}",
+    role=concourse_instance_role.name,
+    path="/ol-applications/concourse/profile/",
+)
+
+# Mount Vault secrets backend and populate secrets
+concourse_secrets_mount = vault.Mount(
+    "concourse-app-secrets",
+    description="Generic secrets storage for Concourse application deployment",
+    path="secret-concourse",
+    type="kv-v2",
+)
+vault.generic.Secret(
+    "concourse-web-secret-values",
+    path=concourse_secrets_mount.path.apply(lambda mount_path: f"{mount_path}/web"),
+    data_json=concourse_config.require_secret_object("web_vault_secrets").apply(
+        json.dumps
+    ),
+)
+vault.generic.Secret(
+    "concourse-worker-secret-values",
+    path=concourse_secrets_mount.path.apply(lambda mount_path: f"{mount_path}/worker"),
+    data_json=concourse_config.require_secret_object("worker_vault_secrets").apply(
+        json.dumps
+    ),
+)
+vault.generic.Secret(
+    "concourse-dockerhub-credentials",
+    path=concourse_secrets_mount.path.apply(
+        lambda mount_path: f"{mount_path}/concourse/main/dockerhub"
```

Comment:
Doesn't this make the path `secret-concourse/concourse/main/dockerhub`?

Reply:
Yes, that's where the concourse job will look